server/protocols/jmap/src/main/java/org/apache/james/jmap/Version.java

```diff
@@ -0,0 +1,48 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap;
+
+import java.util.Arrays;
+
+import com.google.common.base.Preconditions;
+
+public enum Version {
```

Comment:
If we want the code to be modular (what is intended by using a multibinder on routes implementation versions), we can't use an enum.

Reply:
It's a design choice we have to make.

Reply:
The only corner case is if in the future there's a new JMAP spec version, someone wants to implement and inject it without touching core James code (thus modifiying that enum version).

If you see James as a toolkit, and want to quickly implement that nbrand new JMAP version, then yes, we cannot rely on an enum :+1: .

While this seems overkill at first sight (I hope a new JMAP release to not happen anytime soon), being more composable sounds like a good idea, and might allow some future contribution, thus I believe we can have POJO class + constants for existing versions.

Sorry, @mbaechler , I had to thought a long time to come over with that, was it what you meant?

Reply:
I don't have any opinion to be honest, both are ok. However, keeping a multibinder and using an enum at the same time doesn't make sense.
Let's use constants if you want.

Reply:
Not sure to get what is the conclusion you reached TBH

Enum or not enum?

Reply:
I can contribute that changeset ^^

Reply:
> Enum or not enum?

Not enum

Reply:
We still need to convert that enum to something else

Reply:
So now `Version` is a pojo with constants. Tell me if it's ok !